pyiceberg/table/__init__.py

```diff
@@ -1212,6 +1213,23 @@ def to_daft(self) -> daft.DataFrame:
 
         return daft.read_iceberg(self)
 
+    @staticmethod
+    def metadata_file_location(table_location: str, file_name: str, properties: Properties = EMPTY_DICT) -> str:
+        """Get the full path for a metadata file.
+
+        Args:
+            table_location (str): The base table location
+            file_name (str): Name of the metadata file
+            properties (Properties): Table properties that may contain custom metadata path
+
+        Returns:
+            str: Full path where the metadata file should be stored
+        """
```

Comment:
nit: what if we model it like `new_data_location`
https://github.com/apache/iceberg-python/blob/509713bbc23468008eb9e5ac659db9dbb6828c6d/pyiceberg/table/locations.py#L55-L65

```suggestion
    @staticmethod
    def new_metadata_location(table_location: str, file_name: str, properties: Properties = EMPTY_DICT) -> str:
        """Return a fully-qualified metadata file location for the given filename.

        Args:
            table_location (str): The base table location
            file_name (str): Name of the metadata file
            properties (Properties): Table properties that may contain custom metadata path

        Returns:
            str:  A fully-qualified location URI for the metadata file.
        """
```